nixos/modules/services/misc/paperless.nix

```diff
@@ -160,9 +163,10 @@ in
       description = lib.mdDoc "Web interface port.";
     };
 
-    # FIXME this should become an RFC42-style settings attr
-    extraConfig = mkOption {
-      type = types.attrs;
+    settings = mkOption {
```

Comment:
Why not leave this as
```suggestion
    extraConfig = mkOption {
```
?

Reply:
But does that say it has to be called `settings`?

Reply:
In the first paragraph. I'd encourage you to read it.

Reply:
I guess we read it differently.

Reply:
Even if we didn't explicitly decide on the name in RFC42 (IMHO, it's more than an implicit decision), this style of option is called `settings` all over Nixpkgs. I see no reason to do it differently here.

Reply:
Renaming breaks existing configs for no reason.

Reply:
I think we need to break existing configs anyway, otherwise the builtins.toJSON are not removed.

Reply:
As mentioned in the other thread, I don't compat is broken. toJSONs would become superfluous with this but they'd still work; they're just strings.

Reply:
I changed my config back to extraConfig and to contain a builtins.toJSON and it seems to not cause as much issues as I first thought. So we can use mkRenamedOption.

Reply:
> So we can use mkRenamedOption

In that case we also don't create existing configs but just display a warning, right?